app/controllers/locations_controller.rb

```diff
@@ -41,18 +43,31 @@ def deactivate_confirm
   end
 
   def deactivate
-    @location.run_deactivation_validations!
-
-    if @location.location_deactivation_periods.create!(deactivation_date:) && update_affected_logs
+    if @location.location_deactivation_periods.create!(location_id: @location.id, deactivation_date: params[:deactivation_date]) && reset_location_and_scheme_for_logs!
```

Comment:
```suggestion
    if @location.location_deactivation_periods.create!(deactivation_date: params[:deactivation_date]) && reset_location_and_scheme_for_logs!
```

Can remove `location_id` from here now :)